AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -25,6 +25,7 @@
 static int const MAX_ENDPOINT_ATTRIBUTE_METRIC_KEY_LENGTH = 50;
 static int const MAX_ENDPOINT_ATTRIBUTE_VALUE_LENGTH = 100;
 static int const MAX_ENDPOINT_ATTRIBUTE_VALUES = 50;
+static AWSPinpointEndpointProfileDemographic *customDemographic;
```

Comment:
I'm not thrilled about using a global to store this, and the flow feels weird to me--in order to get this custom behavior, I have to call `setCustomDemographic` before initialization, and if I call it afterwards, it does not change the stored value. It breaks normal OO get/set behavior.

What about an overload of `-[updateEndpointProfileWithContext:]` like `-[updateEndpointProfileWithContext:customDemographic]` that takes a non-nullable demographic and immediately uses it in place of the default? Alternately, what is it about the current code that makes the already-existing setter for `demographic` not work as expected? 

Out of curiosity, how do Android & JS implement this feature? Also, since this PR is primarily to achieve feature parity, is the naming (`set/getCustomDemographic`) consistent with those platforms?

Reply:
My feedback is pretty much aligned with Tim's.

As for the Android/JS implementation, I believe the expectation from the customer in this case and the way those two platforms approach is, is that the API is transparent whether the default or the custom attributes are being sent. Therefore, from the customer perspective there's no new API, the custom values are set directly into the existing endpoint instance and if they are set, the Pinpoint client uses them, otherwise if falls back to the defaults.

Reply:
Yea, to be honest, I was not thrilled about it either, but the main interface to instantiating Pinpoint was this:
```
        let pinpointConfiguration = AWSPinpointConfiguration.defaultPinpointConfiguration(launchOptions: launchOptions)
        let pinpoint = AWSPinpoint(configuration: pinpointConfiguration)
```

If we try to do something like:
```
let pinpointConfiguration = AWSPinpointConfiguration.defaultPinpointConfiguration(launchOptions: launchOptions)
let pinpoint = AWSPinpoint(configuration: pinpointConfiguration)
pinpoint.targetingClient.currentEndpointProfile().demographic = customDemographic
```
I noticed that it is possible that it is possible that we can send out an event without the customDemographic fields.  This leads me to believe that by creating an overload of `updateEndpointProfileWithContext:customDemographic:` in the targeting client will not work.  That being said, I'll give it try and triple check.

Also, I don't understand what you mean by `and if I call it afterwards, it does not change the stored value.`.  I believe that if you do call it at a later time, it will change the value being sent out.

Android & JS -- i should have dove deeper.. I didn't in this case.  i'll get back to you.

Reply:
Spoke to @palpatim offline about this.  Moving away from the initially proposed static variable.  Concerns about sending out the default demographic first and then sending out a custom one should not be of a concern since the backend should handle updating an endpoint if the custom demographics are updated for that endpoint.  I verified in the pinpoint console and only customDemographics show up. :)